psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -104,13 +104,58 @@ def scf_compute_energy(self):
 def scf_initialize(self):
     """Specialized initialization, compute integrals and does everything to prepare for iterations"""
 
+    # Figure out memory distributions
+
+    # Get memory in terms of doubles
+    total_memory = (core.get_memory() / 8) * core.get_global_option("SCF_MEM_SAFETY_FACTOR")
+
+    # Figure out how large the DFT collocation matrices are
+    vbase = self.V_potential()
+    if vbase:
+        collocation_size = vbase.grid().collocation_size()
+        if vbase.functional().ansatz() == 1:
+            collocation_size *= 4 # First derivs
+        elif vbase.functional().ansatz() == 2:
+            collocation_size *= 10 # Second derivs
+    else:
+        collocation_size = 0
+
+
+    # Change allocation for collocation matrices based on DFT type
+    scf_type = core.get_global_option('SCF_TYPE').upper()
+    nbf = self.get_basisset("ORBITAL").nbf()
+    naux = self.get_basisset("DF_BASIS_SCF").nbf()
+    if "DIRECT" == scf_type:
+        jk_size = total_memory * 0.1
+    elif scf_type.endswith('DF'):
+        jk_size = naux * nbf * nbf
+    else:
+        jk_size = nbf ** 4
+
+    # Give remaining to collocation
+    if total_memory > jk_size:
+        collocation_memory = total_memory - jk_size
+    # Give up to 10% to collocation
```

Comment:
This looks like a good heuristic to me - it safely accounts for JK, which is good.

Reply:
Good deal, I figure we can go back and tune the percentages later if someone feels up for trying a few scenarios.